Source/Phone/Reachability/MediaClusterClient.swift

```diff
@@ -20,12 +20,23 @@
 
 import Foundation
 
-class MediaClusterClient: CompletionHandlerType<MediaCluster> {
+class MediaClusterClient {
+    
+    typealias ObjectHandler = (ServiceResponse<MediaCluster>) -> Void
+    
+    private let authenticationStrategy: AuthenticationStrategy
+    private let deviceService: DeviceService
+    
+    init(authenticationStrategy: AuthenticationStrategy, deviceService: DeviceService) {
+        self.authenticationStrategy = authenticationStrategy
+        self.deviceService = deviceService
+    }
+    
     private func requestBuilder() -> ServiceRequest.Builder {
-        return ServiceRequest.Builder().baseUrl(DeviceService.sharedInstance.getServiceUrl("calliopeDiscovery")!)
+        return ServiceRequest.Builder(authenticationStrategy).baseUrl(deviceService.device!.calliopeDiscoveryServiceUrl)
```

Comment:
Again, each case you see that was already deeply reliant on that information being there but was blowing up trying to force an optional string into a URL. This is better, it's more clear it can blow up, but yes, still needs to be better.